src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int index = random.nextInt(5);
+        ColorSupplier colorSupplier = new ColorSupplier();
+
+        Circle circle = new Circle();
```

Comment:
No reason to create all types, you can just defile Figure figure;
and then figure = new Circle(......